test/test_type.jl

```diff
@@ -0,0 +1,98 @@
+module TestType
+
+using Test
+using Trixi
+
+include("test_trixi.jl")
+
+# Start with a clean environment: remove Trixi.jl output directory if it exists
+outdir = "out"
+isdir(outdir) && rm(outdir, recursive = true)
+
+# Run unit tests for various equations
+@testset "Float32 Type Stability" begin
+    @timed_testset "Acoustic Perturbation 2D" begin
+        v_mean_global = (0.0f0, 0.0f0)
+        c_mean_global = 1.0f0
+        rho_mean_global = 1.0f0
+        equations = AcousticPerturbationEquations2D(v_mean_global, c_mean_global,
+                                                    rho_mean_global)
+
+        x = SVector(0.0f0, 0.0f0)
+        t = 0.0f0
+        u = u_ll = u_rr = SVector(1.0f0, 1.0f0, 1.0f0, 1.0f0, 1.0f0, 1.0f0, 1.0f0)
+        orientations = [1, 2]
+        normal_direction = SVector(1.0f0, 1.0f0)
+
+        @test eltype(initial_condition_constant(x, t, equations)) == Float32
```

Comment:
```suggestion
        @test eltype(@inferred initial_condition_constant(x, t, equations)) == Float32
```
We should also check type stability like this

Reply:
This used for something like `@test @inferred func(c1, c2, c3, ..) == c`. Not necessary here